util/src/network/host.rs

```diff
@@ -583,9 +583,9 @@ impl<Message> Host<Message> where Message: Send + Sync + Clone {
 			return;
 		} else if create_session {
 			self.start_session(token, io);
-			io.update_registration(token).unwrap_or_else(|e| debug!(target: "net", "Session registration error: {:?}", e));
+			return;
```

Comment:
Before this change session and handshake shared the same connection and it just needed to be updated here. Now sessions gets its own connection and handshake's connection needs to be unregistered instead